src/Microsoft.ML.Featurizers/ShortGrainDropper.cs

```diff
@@ -0,0 +1,424 @@
+    /// that don't have that minimum number.
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    ///
+    /// ###  Estimator Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | Yes |
+    /// | Input column data type | Takes in all columns |
+    /// | Output column data type | Same as the original input type. |
+    ///
+    /// The <xref:Microsoft.ML.Transforms.RobustScalerEstimator> is not a trivial estimator and needs training.
+    ///
+    ///
+    /// ]]>
+    /// </format>
+    /// </remarks>
+    /// <seealso cref="ShortGrainDropperExtensionClass.DropShortGrains(TransformsCatalog, string[], UInt32)"/>
+    public sealed class ShortGrainDropperEstimator : IEstimator<ShortDropTransformer>
+    {
+        private Options _options;
+
+        private readonly IHost _host;
+
+        #region Options
+        internal sealed class Options : TransformInputBase
+        {
+
+            [Argument((ArgumentType.MultipleUnique | ArgumentType.Required), HelpText = "List of grain columns", Name = "GrainColumns", ShortName = "grains", SortOrder = 0)]
+            public string[] GrainColumns;
+
+            [Argument(ArgumentType.AtMostOnce | ArgumentType.Required, HelpText = "Minimum number of values required",
+                Name = "MinPoints", ShortName = "minp", SortOrder = 1)]
+            public UInt32 MinPoints;
+        }
+
+        #endregion
+
+        internal ShortGrainDropperEstimator(IHostEnvironment env, Options options)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            _host = Contracts.CheckRef(env, nameof(env)).Register("ShortDropEstimator");
+            _host.CheckValue(options.GrainColumns, nameof(options.GrainColumns), "Grain columns should not be null.");
+            _host.CheckNonEmpty(options.GrainColumns, nameof(options.GrainColumns), "Need at least one grain column.");
+            Contracts.Check(options.MinPoints > 0, "Min points must be greater than 0");
+
+            _options = options;
+        }
+
+        public ShortDropTransformer Fit(IDataView input)
+        {
+            if (!AllGrainColumnsAreStrings(input.Schema, _options.GrainColumns))
+                throw new InvalidOperationException("Grain columns can only be of type string");
+
+            return new ShortDropTransformer(_host, _options, input);
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            // We dont change the schema, we just drop rows. Still validate grain columns are correct type
+            if (!AllGrainColumnsAreStrings(inputSchema, _options.GrainColumns))
+                throw new InvalidOperationException("Grain columns can only be of type string");
+
+            return inputSchema;
+        }
+    }
+
+    public sealed class ShortDropTransformer : ITransformer, IDisposable
```

Comment:
Change to FilterBase

Reply:
This wont work directly. `FilterBase` -> `TransformBase` -> `IDataTransform`. It is basically a wrapper around an IDataView, not around a transformer. I can change it so my `ShorGrainDataView` uses the `FilterBase`, but not the `ShortDropTransformer`.